pkg/alibabacloud/eni/node.go

```diff
@@ -333,9 +333,6 @@ func (n *Node) ReleaseIPs(ctx context.Context, r *ipam.ReleaseAction) error {
 // GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
 // that can be allocated to the instance
 func (n *Node) GetMaximumAllocatableIPv4() int {
-	n.mutex.RLock()
-	defer n.mutex.RUnlock()
-
```

Comment:
I'm not sure this is the correct fix. We call `n.getLimitsLocked` below which assumes `n.mutex` to be read-locked:

https://github.com/cilium/cilium/blob/e95a201ffa54d05d313d048d9b61f043a397c566/pkg/alibabacloud/eni/node.go#L372-L373

Can you elaborate a bit why read-locking is not needed in this case? It looks like the respective code in `pkg/aws/eni` (which this code seems to be derived from) also read-locks in `(*Node).GetMaximumAllocatableIPv4`. If taking the lock is not necessary, we should rename and adjust the godoc for `n *Node).getLimitsLocked`.

Reply:
yes that is not correct . I have updated.